polkadot/runtime/src/parachains.rs

```diff
@@ -137,24 +138,24 @@ impl<T: Trait> Module<T> {
 		<Parachains<T>>::put(parachains);
 	}
 
-	fn set_heads(aux: &<T as Trait>::PublicAux, heads: Vec<CandidateReceipt>) {
-		assert!(aux.is_empty());
-		assert!(!<DidUpdate<T>>::exists(), "Parachain heads must be updated only once in the block");
-		assert!(
+	fn set_heads(aux: &<T as Trait>::PublicAux, heads: Vec<CandidateReceipt>) -> Result {
+		ensure!(aux.is_empty(), "set_heads must not be signed");
+		ensure!(!<DidUpdate<T>>::exists(), "Parachain heads must be updated only once in the block");
+		ensure!(
 			<system::Module<T>>::extrinsic_index() == T::SET_POSITION,
-			"Parachain heads update extrinsic must be at position {} in the block",
-			T::SET_POSITION
+			"Parachain heads update extrinsic must be at position {} in the block"
+//			, T::SET_POSITION
```

Comment:
Left in for completeness, but given we're returning `&'static str` (for efficiency) and have no formatting tools (since it would severely bloat the runtime), I doubt there's much that can be done right now. A later PR could perhaps switch `&'static str` to `String` in the case of compiling to native and then it can be formatted properly.